sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -650,10 +651,22 @@ private[sql] object DataSourceV2Strategy extends Logging {
 /**
  * Get the expression of DS V2 to represent catalyst predicate that can be pushed down.
  */
-object PushablePredicate {
-  def unapply(e: Expression): Option[Predicate] =
-    new V2ExpressionBuilder(e, true).build().map { v =>
+object PushablePredicate extends Logging {
+
+  def unapply(e: Expression): Option[Predicate] = {
+    val exprOpt = new V2ExpressionBuilder(e, true).build()
+
+    val modifiedExprOpt =
+      if (SQLConf.get.getConf(SQLConf.DATA_SOURCE_DONT_ASSERT_ON_PREDICATE) &&
+        exprOpt.isDefined &&
+        !exprOpt.get.isInstanceOf[Predicate]) {
+        logWarning(log"Predicate expected but got class: ${MDC(EXPR, exprOpt.get.describe())}")
+        None
+      } else { exprOpt }
```

Comment:
the indentation is not right.  See the guideline 

https://github.com/databricks/scala-style-guide#spacing-and-indentation

Reply:
I fail see whats wrong here ?
Is its two space identation for after codeblock within if - right ?
and 2 spaces for each new line, and 2 spaces for each if condition ?

Are you referring to else branch should be in new line ?

Reply:
In our code base, the suggested indentation should be like this 

```
} else { 
  exprOpt 
}
```

Reply:
Got it!

Doc you shared is not specific around this tho. Even has different example:
<img width="1145" alt="image" src="https://github.com/user-attachments/assets/43c9a0c0-0733-4ecd-b25b-68abadea06b3">